drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -255,12 +260,13 @@ static int nrf5_tx(struct device *dev,
 		    nrf5_radio->channel,
 		    nrf5_radio->txpower);
 
-	/* The nRF driver guarantees that either
-	 * nrf_drv_radio802154_transmitted() or
-	 * nrf_drv_radio802154_energy_detected()
-	 * callback is called, thus unlocking the semaphore.
-	 */
-	k_sem_take(&nrf5_radio->tx_wait, K_FOREVER);
+	/* Wait for ack to be received */
+	if (k_sem_take(&nrf5_radio->tx_wait, ACK_TIMEOUT)) {
+		SYS_LOG_DBG("ACK not received");
```

Comment:
Is nrf_drv_radio802154_transmit() handling ACK by itself when the ack flag is set in the frame?

Or, if not, have you had any issues with ACK handling?

Currently, tx is just meant to transmit "and forget", because it's up to the Soft-MAC to handle the ACK. Thus the ieee802154_radio_handle_ack() being called in nrf5_rx_thread().

Reply:
Yes, nrf_drv_radio802154_transmit() is handling ACK by itself

Reply:
Ok then this patch won't work with current 15.4 L2. You'll have to wait for this PR https://github.com/zephyrproject-rtos/zephyr/pull/4094 to get in, so you'll be able to add an HW caps for TX ACK, so the radio strategy driver (ALOHA or CCMA) will know it does not need to wait for the ack by itself.

That said, I wonder how this driver has been working so far, without your patch then.